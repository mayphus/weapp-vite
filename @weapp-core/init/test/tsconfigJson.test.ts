import { getDefaultTsconfigJson, getDefaultTsconfigNodeJson } from '@/tsconfigJson'

describe('tsconfigJson', () => {
  it('getDefaultTsconfigJson', () => {
    expect(getDefaultTsconfigJson()).toMatchSnapshot()
  })

  it('getDefaultTsconfigNodeJson', () => {
    expect(getDefaultTsconfigNodeJson([])).toMatchSnapshot()
  })
})
