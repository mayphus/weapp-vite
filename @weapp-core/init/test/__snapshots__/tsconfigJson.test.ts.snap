// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tsconfigJson > getDefaultTsconfigJson 1`] = `
{
  "compilerOptions": {
    "allowImportingTsExtensions": true,
    "allowJs": true,
    "baseUrl": ".",
    "isolatedModules": true,
    "jsx": "preserve",
    "lib": [
      "ES2020",
      "DOM",
      "DOM.Iterable",
    ],
    "module": "ESNext",
    "moduleResolution": "bundler",
    "noEmit": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "paths": {
      "@/*": [
        "./*",
      ],
    },
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "strict": true,
    "target": "ES2020",
    "types": [
      "miniprogram-api-typings",
    ],
    "useDefineForClassFields": true,
  },
  "exclude": [
    "node_modules",
    "dist",
  ],
  "include": [
    "**/*.ts",
    "**/*.js",
    "vite-env.d.ts",
  ],
  "references": [
    {
      "path": "./tsconfig.node.json",
    },
  ],
}
`;

exports[`tsconfigJson > getDefaultTsconfigNodeJson 1`] = `
{
  "compilerOptions": {
    "allowSyntheticDefaultImports": true,
    "composite": true,
    "module": "ESNext",
    "moduleResolution": "bundler",
    "skipLibCheck": true,
    "strict": true,
  },
  "include": [],
}
`;
